src/importexport/musicxml/internal/musicxml/export/exportmusicxml.cpp

```diff
@@ -8781,16 +8781,83 @@ void ExportMusicXml::harmony(Harmony const* const h, FretDiagram const* const fd
         const String textName = h->hTextName();
         switch (h->harmonyType()) {
         case HarmonyType::NASHVILLE: {
-            m_xml.tag("function", h->hFunction());
-            m_xml.tag("kind", { { "text", textName } }, "none");
+            String alter;
+            String functionText = h->hFunction();
+            if (functionText.empty()) {
+                // we just dump the text as deprecated function
+                m_xml.tag("function", textName);
+                m_xml.tag("kind", "none");
+                break;
+            } else if (!functionText.at(0).isDigit()) {
+                alter = functionText.at(0);
+                functionText = functionText.at(1);
+            }
+            m_xml.startElement("numeral");
+            m_xml.tag("numeral-root", functionText);
+            if (alter == u"b") {
+                m_xml.tag("numeral-alter", "-1");
+            } else if (alter == u"#") {
+                m_xml.tag("numeral-alter", "1");
+            }
+            m_xml.endElement();
+            if (!h->xmlKind().isEmpty()) {
+                String s = u"kind";
+                String kindText = h->musicXmlText();
+                if (!h->musicXmlText().empty()) {
+                    s += u" text=\"" + kindText + u"\"";
+                }
+                if (h->xmlSymbols() == "yes") {
+                    s += u" use-symbols=\"yes\"";
+                }
+                if (h->xmlParens() == "yes") {
+                    s += u" parentheses-degrees=\"yes\"";
+                }
+                m_xml.tagRaw(s, h->xmlKind());
+            } else {
+                // default is major
+                m_xml.tag("kind", "major");
+            }
         }
         break;
         case HarmonyType::ROMAN: {
-            // TODO: parse?
-            m_xml.tag("function", h->hTextName());   // note: HTML escape done by tag()
-            m_xml.tag("kind", { { "text", "" } }, "none");
+            int alter = 0;
+            static const std::wregex roman(L"(b|#)?([ivIV]+)");
+            if (textName.contains(roman)) {
+                StringList matches = textName.search(roman, { 1, 2 });
```

Comment:
Can someone (@cbjeukendrup?) explain to me what exactly this does? I don't understand the code in src/framework/global/types/string.cpp, and esp. not how to do this with plain Qt methods

Reply:
It looks for the parts of the regex (in parentheses) and stores them in the string list.

Reply:
Yes. But this could surely be done in other ways.

Reply:
I've now resorted to use `mid()` in the (potential) 2nd part of the expression, IMHO a lot simpler